gips/0083.md

```diff
@@ -0,0 +1,240 @@
+  provider:
+    id: my-company
+    name: My Company
+    logo: https://company.example.com/logo.png
+```
+
+The service `substreams-v1` means that the endpoint makes available a gRPC endpoint responding to these methods:
+
+- sf.substreams.rpc.v2.Stream/Block
+- sf.substreams.rpc.v2.EndpointInfo/Info
+
+as specified by https://github.com/streamingfast/substreams/blob/develop/proto/sf/substreams/rpc/v2/service.proto updated from time-to-time in backwards compatible ways.
+
+The Payment Gateway MUST listen to the Arbitrum Staking contract for such registrations (through a Substreams module streamed, a Subgraph like the canonical Network Subgraph or other means), and update its local view of the network.
+
+The Payment Gateway CAN have systems to perform health checks (e.g. on `/health` or `/info` endpoints), or other checks to ensure active, up-to-date and properly configured providers are offered to end users. A Payment Gateway CAN check block height to ensure the backing provider is close to chain head before offering it to users.
+
+
+## Indexer Selection Algorithm (ISA)
+
+To provide consumers with fair access to Substreams providers, an Indexer Selection Algorithm is needed.
+
+Given that there's no gateway involved with Substreams (it's direct point-to-point between consumer and provider), the existing Indexer Selection Algorithm (ISA), which happens at the time of query, cannot be applied for Substreams.  Therefore, a round-robin selection of providers is offered on the front-end of the Payment Gateway (https://thegraph.market), ensuring fair distribution of request loads, while still allowing consumers to choose their provider.
+
+> NON-NORMATIVE: A reason for selecting a provider could be that it has pre-cached data for a given dataset, which would speed up and lower the costs for a consumer.
+
+> NON-NORMATIVE: Future GIPs will propose improved ISA, based on better data and better metrics, which might be desirable to consumers.
+
+## Payments
+
+Payments are handled by the Payment Gateway, which is responsible for collecting payments from consumers and distributing them to indexers.
+
+> INFORMATIVE: The Graph Market acts as the first Payments Gateway. Such a gateway DOES NOT route traffic, but does route payments and receive consumption signals from Indexers.
+
+Such a Payment Gateway is responsible for invoking the `collect` function in the staking contract, thereby transfering funds from Consumers to Indexers.
+
+This builds on work done in [GIP-0056 Permissionless Payers](https://github.com/graphprotocol/graph-improvement-proposals/blob/main/gips/0056-permissionless-payers.md), which opened up an opportunity for such new data services to use the network for payment.
+
+> INFORMATIVE: [TAP](https://docs.rs/tap_core/latest/tap_core/index.html) is not required for Substreams to honor the promises of the protocol. However, as TAP brings augmented trust minimization properties, it would be incorporated in future work.
```

Comment:
We've renamed TAP to GraphTally! Could you update the name here?

Reply:
Can we talk a bit about how you're thinking we would use GraphTally here? Currently, the way things work is that gateways send signed receipts to Indexers. Indexers then send a batch of those signed receipts back to the gateway to get a signed RAV (Aggregate receipt) which Indexers then post onchain to release money from the escrow contract. It sounds to me like with substreams, Indexers and Consumers talk directly, so gateways aren't really generating receipts. Is there an inversion of trust that happens here compared to the subgraph flow? Payment gateways trust Indexers are accurately reporting usage?